api/api.go

```diff
@@ -155,6 +155,8 @@ type Config struct {
 	//
 	// TLSConfig is ignored if HttpClient is set.
 	TLSConfig *TLSConfig
+
+	Headers map[string]string
```

Comment:
How do you feel about setting this to be an [`http.Header`](https://golang.org/pkg/net/http/#Header) as well? It would help with multi-valued headers, and may eliminate the need to copy below.